src/acc/libsmm_acc/libcusmm/predict_train.py

```diff
@@ -0,0 +1,1165 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+####################################################################################################
+# Copyright (C) by the DBCSR developers group - All rights reserved                                #
+# This file is part of the DBCSR library.                                                          #
+#                                                                                                  #
+# For information on the license, see the LICENSE file.                                            #
+# For further information please visit https://dbcsr.cp2k.org                                      #
+# SPDX-License-Identifier: GPL-2.0+                                                                #
+####################################################################################################
+
+import os
+import sys
+import datetime
+import json
+import random
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from optparse import OptionParser
+from predict_helpers import (
+    safe_pickle,
+    safe_pickle_load,
+    plot_training_data,
+    plot_choice_goodness,
+    plot_performance_gains,
+    plot_scaled_performance_gains,
+    plot_absolute_performance_gain,
+    plot_relative_performance_gain,
+    performance_gain,
+)
+from kernels.cusmm_predict import to_tuple, to_string
+
+
+# ===============================================================================
+def main():
+    """
+    This script is part of the workflow for predictive modelling of optimal libcusmm parameters.
+    For more details, see predict.md
+
+    """
+
+    parser = OptionParser()
```

Comment:
while still working, [`OptionParser`](https://docs.python.org/2/library/optparse.html) was deprecated in Python 2.7 and got replaced with [argparse](https://docs.python.org/3/library/argparse.html).

Reply:
Thanks!
`OptionParser` is not used just here though, but throughout python scripts in DBCSR.
I'll change that ...